Kudu.Core/Deployment/FetchDeploymentManager.cs

```diff
@@ -213,6 +223,14 @@ public async Task PerformDeployment(DeploymentInfoBase deploymentInfo, IDisposab
                                 deploymentInfo: deploymentInfo,
                                 needFileUpdate: deploySpecificCommitId,
                                 fullBuildByDefault: deploymentInfo.DoFullBuildByDefault);
+
+                            if (updateStatusObj != null)
+                            {
+                                // Since build is success we need to force restart on windows
+                                // Directly set this instead of Constants.BuildSuccessful
+                                updateStatusObj.DeploymentStatus = Constants.PostBuildRestartRequired;
```

Comment:
We should do this from RestartMainSiteIfNeeded() by substituting old logic. But because Windows flow is not fully implemented, hide it behind an app setting for now.

@jvano @JasonFreeberg  - FYI

Reply:
Notes from 10/12 meeting concerning the above:

  - Adjust the hook to check for the custom deployment ID for code apps. If the custom deployment ID exists, then we don't do the restart callback
  - In that case we know that the client is using the new deployment flow, and the restart will be handled at controller